### PR TITLE
Fixed Issue #18438: Fixed Show popups, wrong text in inherit-button

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

File: application/views/themeOptions/options_core.php [hunk 1]
```diff
@@ -121,8 +121,13 @@
                                 $optionsValues = !empty($attribute['options']) ? explode('|', $attribute['options']) : array();
                                 $optionLabels = !empty($attribute['optionlabels']) ? explode('|', $attribute['optionlabels']) : array();
                                 $options = array_combine($optionsValues, $optionLabels);
-                                if ($bInherit && isset($sParentOption)){
-                                    $options['inherit'] = gT("Inherit") . ' [' . $sParentOption . ']';
+                                if ($bInherit && isset($sParentOption)) {
+                                    if(is_numeric($sParentOption) && array_key_exists($sParentOption, $options)) {
```
Comment (Shnoulle) [inline application/views/themeOptions/options_core.php:125]: Why is_numeric ?

**Review by Shnoulle — APPROVED** [review]
No code issue, but why only to is_numeric ? 'options' => `['Y' => 'Yes']` for example

File: application/views/themeOptions/options_core.php (hunk 1 above)
Comment (gabrieljenik) [reply application/views/themeOptions/options_core.php:125]: That;s how we expect the `sParentOption` to come like.
If it comes like that, we map it. If not, we show it as it is coming.

**Comment by gabrieljenik** [comment]
> No code issue, but why only to is_numeric ? 'options' => `['Y' => 'Yes']` for example

For this particular attribute? I believe they are being stored like numbers here.

File: application/views/themeOptions/options_core.php (hunk 1 above)
Comment (Shnoulle) [reply application/views/themeOptions/options_core.php:125]: For Show popusp : OK https://github.com/LimeSurvey/LimeSurvey/blob/096476f68a8edf1f2eea9bce4ba5817ef6b09238/themes/survey/vanilla/config.xml#L63

But for all other : https://github.com/LimeSurvey/LimeSurvey/blob/096476f68a8edf1f2eea9bce4ba5817ef6b09238/themes/survey/vanilla/config.xml#L65 

We need to translate too …

**Comment by Shnoulle** [comment]
> > No code issue, but why only to is_numeric ? 'options' => `['Y' => 'Yes']` for example
> 
> For this particular attribute? I believe they are being stored like numbers here.

Yes, this one , but you really don't fix on/off or Y/N etc … ?

**Comment by Shnoulle** [comment]
I don't know what to say when it's totally obvious by reading the code...
Capture d’écran du 2023-02-10 09-06-02

It must be **Hérite (Non)**

**Comment by gabrieljenik** [comment]
> I don't know what to say when it's totally obvious by reading the code...

Haven't seen that... Maybe it is not that obvious to everyone
Will do another ticket for that

**Comment by gabrieljenik** [comment]
Created
18633: Theme Options: Inherited value doesn't match any attribute option
https://bugs.limesurvey.org/view.php?id=18633

**Comment by Shnoulle** [comment]
> Created 18633: Theme Options: Inherited value doesn't match any attribute option https://bugs.limesurvey.org/view.php?id=18633

In my opinion it's clearly a duplicate …

**Comment by gabrieljenik** [comment]
> > Created 18633: Theme Options: Inherited value doesn't match any attribute option https://bugs.limesurvey.org/view.php?id=18633
> 
> In my opinion it's clearly a duplicate …

Let it be , please? .... :)

File: application/views/themeOptions/options_core.php (hunk 1 above)
Comment (olleharstedt) [reply application/views/themeOptions/options_core.php:125]: Another reason to not use hungarian notation, lol. `$sParentOption`, s = string, but is_numeric check? ^^

File: application/views/themeOptions/options_core.php [hunk 2]
```diff
@@ -121,8 +121,13 @@
...
+                                        $sParentLabelOption = $options[$sParentOption];
+                                        $options['inherit'] = gT("Inherit") . ' [' . gT($sParentLabelOption) . ']';
+                                    } else {
+                                        $options['inherit'] = gT("Inherit") . ' [' . gT($sParentOption) . ']';
```
Comment (olleharstedt) [inline application/views/themeOptions/options_core.php:129]: Bug here if sParentOption is empty, it seems.

Comment (olleharstedt) [reply application/views/themeOptions/options_core.php:129]: Tim fixed it in latest release.

Comment (olleharstedt) [reply application/views/themeOptions/options_core.php:129]: How was this tested, @gabrieljenik ???

Comment (gabrieljenik) [reply application/views/themeOptions/options_core.php:129]: I am noticing it. I guess that the tested case was the bug one. Sorry

Comment (gabrieljenik) [reply application/views/themeOptions/options_core.php:129]: On the else part of the if statement, we were supposed to leave the original statement:
`$options['inherit'] = gT("Inherit") . ' [' . $sParentOption . ']';`

**Pull request merged**